sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -2543,6 +2543,21 @@ pi_result cuda_piQueueGetInfo(pi_queue command_queue, pi_queue_info param_name,
   case PI_QUEUE_INFO_PROPERTIES:
     return getInfo(param_value_size, param_value, param_value_size_ret,
                    command_queue->properties_);
+  case PI_QUEUE_INFO_STATUS: {
+    bool IsReady = command_queue->all_of([](CUstream s) -> bool {
+      const CUresult ret = cuStreamQuery(s);
+      if (ret == CUDA_SUCCESS)
+        return true;
+
+      if (ret == CUDA_ERROR_NOT_READY)
+        return false;
+
+      PI_CHECK_ERROR(ret);
```

Comment:
This may throw an exception. Could you please clarify where it is expected to be caught?

Reply:
Thanks for catching this, fixed.